src/include/common/object_pool.h

```diff
@@ -55,12 +70,25 @@ class AlignedByteAllocator {
 template <typename T, class Allocator = AlignedByteAllocator<T>>
 class ObjectPool {
  public:
-  /**
+  /***
    * Initializes a new object pool with the supplied limit to the number of
    * objects reused.
-   * @param reuse_limit
+   *
+   * @param size_limit the maximum number of objects the object pool controls
+   * @param reuse_limit the maximum number of reusable objects, which needs be to
+   * not greater than size_limit
    */
-  explicit ObjectPool(const uint64_t reuse_limit) : reuse_limit_(reuse_limit) {}
+  explicit ObjectPool(const uint64_t size_limit, const uint64_t reuse_limit)
```

Comment:
no need for `explicit` keyword since the constructor takes in more than 1 parameter.

Reply:
First, in addition to tli2's observation, I don't see why explicit is deemed necessary here. What breaks if is invoked with a compatible type?

Second, if it is required that reuse_limit <= size_limit, don't just document, enforce it with an assertion or other check and exception. I think assertion appropriate in this case.